### ISSUE TITLE
[codespaces] New repository does not detect presence of new .devcontainer/ files on azd init

### ISSUE THREAD

**Issue by danieljurek** [Azure/azure-dev#964]
Repro steps: 

1. New repository (must have some file to get to "codespaces" experience on GitHub)
2. Launch codespaces in the repo (it doesn't have a template in it yet and no .devcontainer folder) 
3. Install `azd` 
4. Run `azd init` select a template, commit, push 

Expected:
vscode inside of codespaces warns about changes to `.devcontainer` and offers to rebuild container

Actual: 
no warning observed

Workarounds: 
* Rebuild container manually after `azd init` commit, push
* Use a template for your repo so `.devcontainer` is already set up

**Comment by vhvb1989** [comment]
> Does is resolve if we restart/reload Codespaces?

Yes, if you do `F1 + Reload window`, VSCode would detect the changes.

But, IMHO, this the flow mentioned on this issue shouldn't be the common/expected way for using `codespaces`.  This use case is asuming that, as a user, I would start by creating a new repository to launch `codespaces`, and it should be more common to start by cloning or forking a template (to create the new repo) and then just open that repo with `codespaces`.

There's also the case where I have an existing repo and I want to azd-fy it, but in that case, there's unlikely to think that a user would pick a template to be pulled into the existing repo (as the repo might already contain a .vscode folder and other files).

So, in general, I would close this issue advising folks to start by forking an azd template (either using fork or by using the `use template option`) and then opening it with codespaces

**Comment by danieljurek** [comment]
@bwateratmsft -- Can you point us to someone in vscode who can help here?

**Comment by bwateratmsft** [comment]
@chuxel @chrmarti can you take a look? Thanks!

**Comment by joshspicer** [comment]
We'll take a look.  We have a watcher that should provide that 'Prompt for rebuild' when files are added to the .devcontainer folder. 👍

**Comment by savannahostrowski** [comment]
@joshspicer Have you had a chance to investigate this at all? Happy to transfer to the remote containers repo if that makes more sense!
